apps/els_lsp/src/els_code_lens_suggest_spec.erl

```diff
@@ -66,13 +70,27 @@ pois(Document) ->
 %%==============================================================================
 %% Internal functions
 %%==============================================================================
--spec get_type_spec(poi(), els_typer:info() | 'no_info') -> binary().
+-spec get_type_spec(poi(), els_typer:info()) -> binary().
 get_type_spec(POI, Info) ->
-  case Info of
-    'no_info' ->
-      <<"Cannot extract specs (check logs for details)">>;
-    _ ->
-      #{id := {Function, Arity}} = POI,
-      Spec = els_typer:get_type_spec(Function, Arity, Info),
-      re:replace(Spec, ",", ", ", [global, {return, binary}])
+    #{id := {Function, Arity}} = POI,
+    Spec = els_typer:get_type_spec(Function, Arity, Info),
+    re:replace(Spec, ",", ", ", [global, {return, binary}]).
+
+-spec truncate_spec_title(binary(), number()) -> binary().
+truncate_spec_title(Spec, MaxLength) ->
+  {Encoding, _BOMLength} = unicode:bom_to_encoding(Spec),
+  CodedSpec = unicode:characters_to_list(Spec, Encoding),
+  Length = string:length(CodedSpec),
+  case Length > MaxLength of
+    true ->
+      Title = unicode:characters_to_binary(
+        string:slice(CodedSpec, 0, MaxLength - 3)
+      ),
+      <<Title/binary, "...">>;
+    false ->
+      Spec
   end.
+
+-spec spec_title_max_length() -> number().
```

Comment:
nitpick: can max length be also `float()` or only `integer`?

Reply:
fixed